From be156caff9b57a45bb1f9e4491b1d40e47675b67 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Fri, 16 Mar 2012 12:34:33 +0100
Subject: [PATCH] patch: gimp-2.8

Squashed commit of the following:

commit 5c50f3be8e1f665e4a7b071b09738e00a1f6adb1
Author: nkbj <nkbj>
Date:   Sat Dec 3 05:30:20 2011 +0000

    Update code for GIMP 2.8 (tested with GIMP 2.7.3).
    (cherry picked from commit 681d2e46e0b0fbcc601068025cf3dbcd3c5bdb9b)
---
 ufraw-gimp.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/ufraw-gimp.c b/ufraw-gimp.c
index c393336..b5ed51c 100644
--- a/ufraw-gimp.c
+++ b/ufraw-gimp.c
@@ -361,7 +361,15 @@ long ufraw_save_gimp_image(ufraw_data *uf, GtkWidget *widget)
     layer = gimp_layer_new(uf->gimpImage, _("Background"), Crop.width,
                            Crop.height, depth == 3 ? GIMP_RGB_IMAGE : U16_RGB_IMAGE,
                            100.0, GIMP_NORMAL_MODE);
+#ifdef UFRAW_CINEPAINT
+    gimp_image_add_layer(uf->gimpImage, layer, 0);
+#else
+#if defined(GIMP_CHECK_VERSION) && GIMP_CHECK_VERSION(2,7,2)
+    gimp_image_insert_layer(uf->gimpImage, layer, 0, 0);
+#else
     gimp_image_add_layer(uf->gimpImage, layer, 0);
+#endif
+#endif
 
     /* Get the drawable and set the pixel region for our load... */
     drawable = gimp_drawable_get(layer);
-- 
1.7.7.6

